admin-ui/config/webpack.config.client.prod.ts

```diff
@@ -36,6 +36,63 @@ const webpackConfig: WebpackConfig & { devServer?: DevServerConfig } = {
   optimization: {
     splitChunks: {
       chunks: 'all',
+      cacheGroups: {
+        // Vendor libraries - split by framework/library type
+        react: {
+          test: /[\\/]node_modules[\\/](react|react-dom|react-router|react-router-dom)[\\/]/,
+          name: 'react-vendor',
+          chunks: 'all',
+          priority: 20,
+        },
+        mui: {
+          test: /[\\/]node_modules[\\/](@mui|@emotion)[\\/]/,
+          name: 'mui-vendor',
+          chunks: 'all',
+          priority: 19,
+        },
+        redux: {
+          test: /[\\/]node_modules[\\/](@reduxjs|redux|redux-saga|redux-persist)[\\/]/,
+          name: 'redux-vendor',
+          chunks: 'all',
+          priority: 18,
+        },
+        charts: {
+          test: /[\\/]node_modules[\\/](recharts|react-ace|ace-builds)[\\/]/,
+          name: 'charts-vendor',
+          chunks: 'all',
+          priority: 17,
+        },
+        utils: {
+          test: /[\\/]node_modules[\\/](lodash|moment|dayjs|axios|formik|yup)[\\/]/,
+          name: 'utils-vendor',
+          chunks: 'all',
+          priority: 16,
+        },
+        // Plugin chunks - each plugin gets its own chunk
+        plugins: {
+          test: /[\\/]plugins[\\/]/,
+          name: (module: any) => {
+            const pluginName = module.context.match(/[\\/]plugins[\\/]([^\\/]+)[\\/]/)?.[1]
+            return pluginName ? `plugin-${pluginName}` : 'plugin-common'
+          },
+          chunks: 'all',
+          priority: 15,
+        },
```

Comment:
Guard `module.context` before calling `.match`.

`module.context` can be `undefined` (e.g., for concatenated, external, or virtual modules). Calling `.match` on `undefined` will crash the production build. Please add a null check/optional chaining before invoking `.match`. 

```diff
-        plugins: {
+        plugins: {
           test: /[\\/]plugins[\\/]/,
-          name: (module: any) => {
-            const pluginName = module.context.match(/[\\/]plugins[\\/]([^\\/]+)[\\/]/)?.[1]
+          name: (module: any) => {
+            const pluginName = module?.context?.match(/[\\/]plugins[\\/]([^\\/]+)[\\/]/)?.[1]
             return pluginName ? `plugin-${pluginName}` : 'plugin-common'
           },
```

Reply:
done